addons/mail/static/tests/legacy/core/record_tests.js

```diff
@@ -690,3 +690,47 @@ QUnit.test("record list assign should update inverse fields", async (assert) =>
     jane.thread = [];
     assert.ok(jane.notIn(general.members));
 });
+
+QUnit.test("datetime type record", async (assert) => {
+    (class Thread extends Record {
+        static id = "name";
+        name;
+        date = Record.attr(undefined, { type: "datetime" });
+    }).register();
+    const store = await start();
+    const general = store.Thread.insert({ name: "General", date: "2024-02-20 14:42:00" });
+    assert.ok(general.date instanceof luxon.DateTime);
+    assert.ok(general.date.day === 20);
+    general.date = "2024-02-21 14:42:00";
+    assert.ok(general.date.day === 21);
+    general.date = undefined;
+    assert.ok(general.date === undefined);
+});
+
+QUnit.test("Datetime type shoud not trigger onUpdate when the date is the same", async () => {
+    (class Thread extends Record {
+        static id = "name";
+        name;
+        date = Record.attr(undefined, {
+            type: "datetime",
+            onUpdate: () => step("DATE_UPDATED"),
+        });
+    }).register();
+    const store = await start();
+    store.Thread.insert({ name: "General", date: "2024-02-20 14:42:00" });
+    store.Thread.insert({ name: "General", date: "2024-02-20 14:42:00" });
+    await assertSteps(["DATE_UPDATED"]);
```

Comment:
`assertSteps` should be between each line of the test to have full coverage, this doesn't guarantee which step is triggering it and which is not